src/render/Renderer.cpp

```diff
@@ -2152,26 +2152,32 @@ bool CHyprRenderer::applyMonitorRule(CMonitor* pMonitor, SMonitorRule* pMonitorR
 void CHyprRenderer::setCursorSurface(wlr_surface* surf, int hotspotX, int hotspotY, bool force) {
     m_bCursorHasSurface = surf;
 
-    if ((surf == m_sLastCursorData.surf || m_bCursorHidden) && hotspotX == m_sLastCursorData.hotspotX && hotspotY == m_sLastCursorData.hotspotY && !force)
+    if (surf == m_sLastCursorData.surf && hotspotX == m_sLastCursorData.hotspotX && hotspotY == m_sLastCursorData.hotspotY && !force)
         return;
 
     m_sLastCursorData.name     = "";
     m_sLastCursorData.surf     = surf;
     m_sLastCursorData.hotspotX = hotspotX;
     m_sLastCursorData.hotspotY = hotspotY;
 
+    if (m_bCursorHidden)
+        return;
+
     wlr_cursor_set_surface(g_pCompositor->m_sWLRCursor, surf, hotspotX, hotspotY);
 }
 
 void CHyprRenderer::setCursorFromName(const std::string& name, bool force) {
     m_bCursorHasSurface = true;
 
-    if ((name == m_sLastCursorData.name || m_bCursorHidden) && !force)
+    if (name == m_sLastCursorData.name && !force)
         return;
 
     m_sLastCursorData.name = name;
     m_sLastCursorData.surf.reset();
 
+    if (m_bCursorHidden)
```

Comment:
honestly for the sake of clarity `force` should also omit this check. I know it won't change anything rn, but in the future it might be annoying